In [ ]:
import numpy as np
import pandas as pd
heroes = pd.read_csv("heroes_information.csv")

In [ ]:
###instalar desde anaconda navigator######
from pandasql import *

In [ ]:
pysqldf = lambda q: sqldf(q,globals())

In [ ]:
query0 = " SELECT * FROM heroes limit 10"

In [ ]:
pysqldf(query0)

#### 1. Cuál es el nombre del primer superhéroe de la tabla heroes?

In [30]:
query1 = " SELECT name FROM heroes limit 1 "

In [31]:
pysqldf(query1)

name
0  A-Bomb

R: El nombre del primer superheroes de la tabla es A-bomb. 

#### 2. Cuántas casas publicadoras existen?

In [32]:
query2 = " SELECT COUNT(DISTINCT Publisher) AS Cant_Publicadores FROM heroes "

In [33]:
pysqldf(query2)


Cant_Publicadores
0                 24

Existen 24 casas publicadoras.

#### 3. Cuántos superhéroes miden más de 2 metros (200 cms)

In [ ]:
query3 = " SELECT COUNT(Height) FROM heroes WHERE Height > 200 "

In [ ]:
pysqldf(query3)

Hay 59 superhéroes que miden más de 2 metros.

###### 4. Cuántos superhéroes son humanos y miden más de 2 metros (200 cms)?

In [12]:
query4 = " SELECT COUNT(*) FROM heroes WHERE Race = 'Human' AND Height > 200 "

In [13]:
pysqldf(query4)

COUNT(*)
0        12

Existen únicamente 12 humanos que midan más de 2 metros.

###### 5. Cuántos superhéroes pesan más de 100 lbs y menos de 200 lbs?

In [14]:
query5 = " SELECT COUNT(Weight) FROM heroes WHERE Weight BETWEEN 100 AND 200"

In [15]:
pysqldf(query5)

COUNT(Weight)
0             98

Hay 98 superhéroes que pesen entre 100 y 200 libras.

###### 6. Cuántos superhéroes tienen los ojos de color azul o rojo?

In [18]:
query6 = " SELECT COUNT(*) FROM heroes WHERE `Eye color` IN ('red', 'blue') "

In [19]:
pysqldf(query6)

COUNT(*)
0       271

Hay 271 superhéroes que tienen hojos de color azul o rojo.

###### 7. Cuántos superhéroes son Human, Mutant y tienen el pelo color Verde o son Vampiros con pelo Negro.

In [20]:
query7 = " SELECT COUNT(*) FROM heroes WHERE (Race='Human' AND `Hair color` = 'Green') OR (Race = 'Mutant' AND `Hair color` = 'Green') OR (Race = 'Vampire' AND `Hair color`= 'Black') "


In [21]:
pysqldf(query7)

COUNT(*)
0         5

Hay solo 5 superhéroes que sean humanos, mutantes y que tengan el pelo verde; o vampiros que tengan el pelo negro.

###### 8. Cuál es el nombre del primer superhéroe si ordeno la tabla por raza en orden descendente?

In [35]:
query8 = " SELECT name FROM heroes ORDER BY Race DESC limit 1"

In [36]:
pysqldf(query8)

name
0  Solomon Grundy

Si ordenamos la tabla por orden descendente, el nombre del primer superhéroe es Solomon Grundy.

###### 9. Cuántos superhéroes son de género masculino y cuántas de género femenino?

In [24]:
query9 = " SELECT Gender,Count(*) FROM heroes GROUP BY Gender HAVING COUNT(gender) > 100 "

In [25]:
pysqldf(query9)

Gender  Count(*)
0  Female       200
1    Male       505

Hay 200 superhéroes de género femenino y 505 de género masculino.

###### 10. Cuántas casas publicadoras tienen más de 15 superhéroes?

In [26]:
query10 = " SELECT Publisher, Count(Publisher) FROM heroes GROUP BY Publisher HAVING COUNT(*) > 15 "

In [27]:
pysqldf(query10)

Publisher  Count(Publisher)
0          DC Comics               215
1  Dark Horse Comics                18
2      Marvel Comics               388
3       NBC - Heroes                19

In [ ]:
###### Hay cuatro casas publicadoras con maa2